source/common/http/conn_manager_utility.cc

```diff
@@ -101,11 +101,11 @@ void ConnectionManagerUtility::mutateRequestHeaders(
     request_headers.insertEnvoyDownstreamServiceCluster().value(config.userAgent().value());
     HeaderEntry& user_agent_header = request_headers.insertUserAgent();
     if (user_agent_header.value().empty()) {
-      user_agent_header.value(config.userAgent().value());
+      user_agent_header.value().setStatic(config.userAgent().value());
     }
 
     if (!local_info.nodeName().empty()) {
-      request_headers.insertEnvoyDownstreamServiceNode().value(local_info.nodeName());
+      request_headers.insertEnvoyDownstreamServiceNode().value().setStatic(local_info.nodeName());
```

Comment:
It's hard to ascertain value without performance measurements or profiles. My gut instinct with most performance things is not to to make things more complicated without this, but you probably have deeper insight into what the hot paths, so if this is on them then let's at least include those changes that might be relevant. I would also include all the ones that are obviously safe, where they reference things that are static only by construction.